geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseConnectionPool.scala

```diff
@@ -61,114 +58,188 @@ object HBaseConnectionPool extends LazyLogging {
+    }
+  }
 
-      if (kerberosTicket.isEmpty) {
-        kerberosTicket = Some(HadoopUtils.kerberosTicketRenewer())
+  /**
+   * Compare two kerberos principals.
+   *
+   * The existing principal is expected to have a realm and hostname filled out already.
+   *
+   * To compare, we: add the default realm, if there is no realm specified; replace _HOST with the current host.
+   *
+   * `SecurityUtil.getServerPrincipal` will replace the _HOST but only if there is already a realm
+   *
+   * @param current existing user
+   * @param principal new user name
+   * @return
+   */
+  private def sameName(current: UserGroupInformation, principal: String): Boolean = {
+    val fullName = if (principal.indexOf('@') != -1) {
+      // we have a realm so this should be work to replace _HOST if present
+      SecurityUtil.getServerPrincipal(principal, null: String)
+    } else {
+      // try to add the default realm and replace _HOST if present
+      Try(KerberosUtil.getDefaultRealm) match {
+        case Success(realm) => SecurityUtil.getServerPrincipal(s"$principal@$realm", null: String)
+        case Failure(e) =>
+          logger.debug(s"Unable to get default Kerberos realm: $e")
+          if (!principal.contains(SecurityUtil.HOSTNAME_PATTERN)) { principal } else {
+            // append a fake realm so that the _HOST replacement works and then remove it afterwards
+            SecurityUtil.getServerPrincipal(s"$principal@foo", null: String).dropRight(4)
+          }
       }
     }
 
-    userCheck = Option(User.getCurrent)
+    current.getUserName == fullName
+  }
 
-    if (last != null && last != userCheck) {
-      logger.warn(s"Detected change in authenticated user from ${last.getOrElse("unauthenticated")} to " +
-          s"${userCheck.getOrElse("unauthenticated")} - this may not work properly with Hadoop security")
-    }
+  /**
+   * Managed connection. The connection itself should not be closed - instead close the wrapper to handle
+   * lifecycle events appropriately.
+   */
+  sealed trait ConnectionWrapper extends Closeable {
+    val connection: Connection
   }
+
+  /**
+   * An unshared connection
+   *
+   * @param connection connection
+   * @param kerberos kerberos ticket renewal thread
+   */
+  case class SingletonConnection(connection: Connection, kerberos: Option[Closeable]) extends ConnectionWrapper {
+    override def close(): Unit = CloseWithLogging(kerberos.toSeq ++ Seq(connection))
+  }
+
+  /**
+   * A shared, cached connection
+   *
+   * @param connection connection
+   * @param kerberos kerberos ticket renewal thread
+   */
+  case class CachedConnection(connection: Connection, kerberos: Option[Closeable]) extends ConnectionWrapper {
+    override def close(): Unit = {}
+  }
+
+  /**
+   * Provided connection - no lifecycle management is performed
+   *
+   * @param connection connection
+   */
+  case class ProvidedConnection(connection: Connection) extends ConnectionWrapper {
+    override def close(): Unit = {}
```

Comment:
Ok, so the Scaladocs indicate that the provider is handling lifecycle management.  Good...